src/fluree/db/query/exec/where.cljc

```diff
@@ -489,15 +495,15 @@
             [s p o]    (assign-matched-values triple solution filters)
             s*         (if (and (matched-iri? s)
                                 (not (get-sid s db-alias)))
-                         (<! (evaluate-subject-iri db error-ch s))
+                         (<! (resolve-subject-sid db error-ch s))
                          s)
             p*         (if (and (matched-iri? p)
                                 (not (get-sid p db-alias)))
-                         (evaluate-predicate-iri db p)
+                         (resolve-predicate-sid db p)
                          p)
             o*         (if (and (matched-iri? o)
                                 (not (get-sid o db-alias)))
-                         (evaluate-predicate-iri db o)
+                         (resolve-predicate-sid db o)
```

Comment:
I saw this when I was doing the rename, should we use `resolve-subject-sid` here instead of `resolve-predicate-sid`? The predicate sid resolution depends on the schema cache, and I'm not 100% certain that the ref iri will always be in there.

I did try it out and got a weird policy test failure, though, so I switched it back.

Reply:
We talked and this is the resolution - this is correct because it is `match-class`, and classes are definitely in the schema cache. So we can safely take this shortcut.